gpflow/quadrature/gauss_hermite.py

```diff
@@ -78,12 +78,15 @@ def __init__(self, dim: int, n_gh: int):
         :param n_gh: number of Gauss-Hermite points, integer
         :param dim: dimension of the multivariate normal, integer
         """
+        print("PRINT - Initializing NDiagGHQuadrature")
+        tf.print("TF PRINT - Initializing NDiagGHQuadrature")
         self.dim = dim
         self.n_gh = n_gh
         self.n_gh_total = n_gh ** dim
-        Z, dZ = ndgh_points_and_weights(self.dim, self.n_gh)
-        self.Z = tf.ensure_shape(Z, (self.n_gh_total, self.dim))
-        self.dZ = tf.ensure_shape(dZ, (self.n_gh_total, 1))
+        with tf.init_scope():
```

Comment:
I've looked at the [docs for `tf.init_scope`](https://www.tensorflow.org/api_docs/python/tf/init_scope) but am still not quite sure why this is needed here - could you briefly explain (perhaps in a code comment) what motivated this introduction?